test/src/service-builders/testSafariDriver.js

```diff
@@ -244,4 +261,121 @@ describe('SafariDriver Transport Tests', function () {
     });
     assert.strictEqual(serverPath, '/Applications/Safari Technology Preview.app/Contents/MacOS/safaridriver');
   });
+
+  it('session create should retry on internal server error (500)', function() {
+    this.timeout(5000);
+
+    let testsPath = [
+      path.join(__dirname, '../../sampletests/async')
+    ];
+
+    MockServer.addMock({
+      url: '/session',
+      statusCode: 500,
+      postdata: JSON.stringify({
+        'capabilities': {'firstMatch': [{}], 'alwaysMatch': {'browserName': 'safari'}}
+      }),
+      response: JSON.stringify({
+        value: {
+          error: 'session not created',
+          message: 'Could not create a session: Some devices were found, but could not be used:\n'
+        }
+      }),
+      times: 2
+    });
+
+    let globals = {
+      reporter(results) {
+        const sep = path.sep;
+        assert.strictEqual(results.errors, 1);
+        assert.strictEqual(Object.keys(results.modules).length, 1);
+        assert.ok(Object.keys(results.modules).includes(`test${sep}sample`));
+        assert.ok(results.lastError instanceof Error);
+
+        assert.strictEqual(results.lastError.message, 'An error occurred while creating a new SafariDriver session: [SessionNotCreatedError] Could not create a session: Some devices were found, but could not be used:\n');
+      }
+    };
+
+    return runTests({source: testsPath}, settings({
+      selenium_host: null,
+      desiredCapabilities: {
+        browserName: 'safari'
+      },
+      webdriver: {
+        host: 'localhost',
+        timeout_options: {
+          retry_attempts: 1
+        }
+      },
+      globals,
+      output: false,
+      output_folder: false
+    }));
+  });
+
+  it('session create should throw error after max retryAttempts', function() {
```

Comment:
Test name seems to be inconsistent with what the test is doing (the test does not throw an error after max retryAttempts but passes on the last attempt).

Reply:
yes you are correct, updated.